exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/FederationGatewaySyncService.java

```diff
@@ -76,27 +76,31 @@ private void doInbound(String date, Optional<String> batchTag) {
 
     private Optional<String> addInboundKeys(String date, Optional<String> batchTag) {
         boolean finished = false;
-        long operationId = operationDao.startOperation(INBOUND);
         Optional<String> localBatchTag = batchTag;
-
-        try {
-            Optional<DownloadData> downloadO = client.download(date, localBatchTag);
-
-            if (downloadO.isPresent()) {
-                DownloadData download = downloadO.get();
-                localBatchTag = Optional.of(download.batchTag);
-                List<TemporaryExposureKey> keys = transform(download.batch);
-                diagnosisKeyDao.addInboundKeys(keys, IntervalNumber.to24HourInterval(Instant.now()));
-                finished = operationDao.finishOperation(operationId, keys.size(), localBatchTag);
-                return download.nextBatchTag;
-
-            } else {
-                return Optional.empty();
-            }
-        } finally {
-            if (!finished) {
-                operationDao.markErrorOperation(operationId, localBatchTag);
+        Optional<Long> operationId = operationDao.startOperation(INBOUND, localBatchTag);
+
+        if (operationId.isPresent()) {
```

Comment:
The if-else could be formatted as
`
return operationId.flatMap(id -> {
  ...
});
`

Or alternatively, even `return operationId.flatMap(this::tryDownload)` where signature is `Optional<DownloadData> tryDownload(long operationId)`

Both options avoid needing to .get() from the optional and the extra -else branch.

Reply:
Yep, I know, for my eye it's just more mess...

Reply:
But I'll check this in some point.

Reply:
Tuned this a little bit. Maybe it's a little bit nicer now. It's good that someone makes noise of these. I must admit that it looked ugly also for me, but not too ugly in current situation. :)

Reply:
It does look cleaner to my eye :) The function now more clearly "does one thing" and the caller decides if it has the sufficient arguments to do make the call in the first place.